src/Angor/Avalonia/Angor.Contexts.Funding/Shared/PublishTransaction.cs

```diff
@@ -25,9 +28,104 @@ public async Task<Result<string>> Handle(PublishTransactionRequest request, Canc
 
             var errorMessage = await indexerService.PublishTransactionAsync(request.TransactionDraft.SignedTxHex);
             
-            return !string.IsNullOrEmpty(errorMessage) 
-                ? Result.Failure<string>(errorMessage) 
-                : Result.Success(request.TransactionDraft.TransactionId);
+            if (!string.IsNullOrEmpty(errorMessage))
+            {
+                return Result.Failure<string>(errorMessage);
+            }
+
+            // If this is an investment draft and we have wallet/project context, persist to portfolio
+            if (request.TransactionDraft is InvestmentDraft investmentDraft && 
```

Comment:
The only time there wouldn't be a record is when the threshold has not been reached, in that case we should get or create and the rest of the code should be the same for all options.

Reply:
I don't fully understand this comment, what does it mean the same for all options? every option updates different params